packages/api/src/routes/project.ts

```diff
@@ -175,17 +175,61 @@ projectRoutes.get('/projects/:projectId/stats', projectIdMiddleware, async (c) =
       llmCount: sum(sql`CASE WHEN requestType = 'llm' THEN 1 ELSE 0 END`).mapWith(Number),
     })
     .from(requestsTable)
-    .where(and(eq(requestsTable.projectId, project.id), gt(requestsTable.timestamp, lastMidnight)));
+    .where(and(eq(requestsTable.projectId, project.id), sql`DATE(timestamp) = CURRENT_DATE`));
 
   const totals = result[0];
 
+  // Get hourly data for the last 24 hours using SQL date functions
+  const hourlyQuery = await db
+    .select({
+      cost: sum(sql`costDetails->'ethGwei'`).mapWith(Number),
+      proxyCount: sum(sql`CASE WHEN requestType = 'proxy' THEN 1 ELSE 0 END`).mapWith(Number),
+      llmCount: sum(sql`CASE WHEN requestType = 'llm' THEN 1 ELSE 0 END`).mapWith(Number),
+      totalTokens: sum(sql`responseDetails->'llmResponse'->'usage'->'total_tokens'`).mapWith(
+        Number
+      ),
+      hour: sql`strftime('%H:%M', timestamp)`.mapWith(String),
+    })
+    .from(requestsTable)
+    .where(
+      and(eq(requestsTable.projectId, project.id), sql`timestamp > datetime('now', '-24 hours')`)
+    )
+    .groupBy(sql`strftime('%H:%M', timestamp)`)
+    .orderBy(sql`strftime('%H:%M', timestamp)`);
+
+  // Create array of hours for the last 24 hours
+  const hours = Array.from({ length: 24 }, (_, i) => {
+    const hour = i.toString().padStart(2, '0');
+    const minute = '00';
+    return `${hour}:${minute}`;
+  });
+
+  // Fill in the data, using 0 for missing hours
+  const hourlyData = hours.map((hour) => {
+    const data = hourlyQuery.find((q) => q.hour === hour);
+    return {
+      timestamp: hour,
+      cost: data?.cost || 0,
```

Comment:
can easily add dummy data by adding a random number instead of 0 here, the current polling behaviour on the frontend means it changes on every request though